pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_conversion.go

```diff
@@ -342,14 +342,34 @@ func (r *HTTPRouteReconciler) gapiToKumaFilter(
 			Percentage: util_proto.Double(100),
 		}
 
-		kumaFilter = &mesh_proto.MeshGatewayRoute_HttpRoute_Filter{
+		kumaFilters = append(kumaFilters, &mesh_proto.MeshGatewayRoute_HttpRoute_Filter{
 			Filter: &mesh_proto.MeshGatewayRoute_HttpRoute_Filter_Mirror_{
 				Mirror: &mirror,
 			},
-		}
+		})
 	case gatewayapi.HTTPRouteFilterRequestRedirect:
 		filter := filter.RequestRedirect
 
+		if p := filter.Path; p != nil {
```

Comment:
Could we extract it to some method? It looks like the code from line 412 is the same.